pkg/services/ngalert/state/historian/loki.go

```diff
@@ -377,6 +377,12 @@ func buildLogQuery(query models.HistoryQuery) (string, error) {
 	if query.RuleUID != "" {
 		logQL = fmt.Sprintf("%s | ruleUID=%q", logQL, query.RuleUID)
 	}
+	if query.DashboardUID != "" {
```

Comment:
If we fine these queries getting slow, we might consider adding filters before the json parsing (filtering before parsing is often quicker)

Reply:
i nearly did that here :smile: but figured it'd be cleaner to refactor ALL filters to use this in the future.

right now, our query latencies are pretty good though: https://ops.grafana-ops.net/d/c152f46c-25dd-4258-8d7c-95298258b683/state-history?orgId=1&refresh=30s&viewPanel=30